components/eam/src/chemistry/mozart/mo_sethet.F90

```diff
@@ -588,13 +667,17 @@ subroutine sethet( het_rates, press, zmid,  phis, tfld, &
           end if
           tmp_hetrates(i,kk,1) = max( 1._r8 / yh2o2,0._r8 ) * stay
           tmp_hetrates(i,kk,2) = max( 1._r8 / yhno3,0._r8 ) * stay
-          xxx3 = (xso2( i,kk) - xgas3(kk))
-          if( xxx3 /= 0._r8 ) then                       ! if no washout lifetime = 1.e29
-             yso2  = xso2( i,kk)/xxx3 * xdtm     
+!++hybrown
+          do jso4 =1,nso4
+          xxx3(jso4) = (xso2( i,kk,jso4) - xgas3(kk,jso4))
+          if( xxx3(jso4) /= 0._r8 ) then                       ! if no washout lifetime = 1.e29
```

Comment:
Just FYI, I know this isn't something you added, but exact floating point comparison like this without tolerance is kind of dangerous; you can never be sure how this logical will evaluate.